lib/ansible/modules/system/user.py

```diff
@@ -726,6 +736,33 @@ def chown_homedir(self, uid, gid, path):
             e = get_exception()
             self.module.exit_json(failed=True, msg="%s" % e)
 
+    def parse_defaults(self):
+        try:
+            f = open('/etc/login.defs', 'r')
+            for line in f:
+                if 'UMASK' in line and '#' not in line.split()[0]:
+                    maskvalue=line.split()[1]
+                    self.newumask = int(maskvalue, 8)
```

Comment:
This would fail with a umask in hexadecimal (using 0x, as mentioned in the file comment) or using regular base 10. Both are unlikely, but better code defensively.